File: wip/ft_transcendence/app/models/channel_message.rb
class ChannelMessage < ApplicationRecord
  enum content_type: [:text, :invite]

  belongs_to :user
  belongs_to :channel

  validates :user, presence: true
  validates :channel, presence: true
  validates :content, presence: true, length: { minimum: 1, maximum: 100 }
  validate_enum_attributes :content_type
end

File: wip/ft_transcendence/frontend/models/ChannelMessage.ts
import { User } from './User'
import { Channel } from './Channel'

export enum ChannelMessageContentType {
  TEXT = 'text',
  INVITE = 'invite',
}

export interface ChannelMessage {
  id: number
  type: ChannelMessageContentType
  content: string
  user: User
  channel: Channel
  createdAt: Date
}
